src/Common/src/Interop/Windows/winhttp/Interop.SafeInternetHandle.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class WinHttp
+    {
+        // TODO: Move SafeHandleZeroOrMinusOneIsInvalid class to Common
+        internal abstract class SafeHandleZeroOrMinusOneIsInvalid : SafeHandle
+        {
+            protected SafeHandleZeroOrMinusOneIsInvalid(bool ownsHandle)
+                : base(IntPtr.Zero, ownsHandle)
+            {
+            }
+
+            public override bool IsInvalid
+            {
+                get { return this.handle == IntPtr.Zero || this.handle == (IntPtr)(-1); }
+            }
+        }
+
+        internal class SafeInternetHandle : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
@bartonjs In my opinion, I don't see any reason to duplicate code and keep re-implementing either of the SafeHandle sub-types regardless of how simple the code is. 
I also like how the names clearly indicate the expectations and make it easier to crosscheck the behavior with documentation of the Win32 APIs.